src/state/mod.rs

```diff
@@ -66,6 +70,36 @@ impl event::EventHandler for MainState {
         graphics::present(ctx)?;
         Ok(())
     }
+
+    fn key_down_event(&mut self, _ctx: &mut Context, keycode: KeyCode, _keymods: KeyMods, _repeat: bool) {
+        match keycode {
+            KeyCode::W => {
+                self.player1.move_up();
+            },
+            KeyCode::S => {
+                self.player1.move_down();
+            },
+            KeyCode::Up => {
+                self.player2.move_up();
+            },
+            KeyCode::Down => {
+                self.player2.move_down();
+            },
+            _ => (),
+        };
+    }
+
+    fn key_up_event(&mut self, _ctx: &mut Context, keycode: KeyCode, _keymods: KeyMods) {
```

Comment:
Doesn't have `key_pressed_event`? this could become one function then

Reply:
The only ones that exist are `key_down_event` and `key_up_event`.

I've searched here and that's all I've found: https://docs.rs/ggez/0.5.1/ggez/event/trait.EventHandler.html

Reply:
ok